### ISSUE TITLE
2FA input fields prompt for autocomplete

### ISSUE THREAD

**Issue by stwalkerster** [enwikipedia-acc/waca#569]
Alpha3031 notes:
> The new 2FA is nice, but having the field set to type="password" means my browser tries to save every single one time code. Maybe set type="text" and autocomplete="one-time-code" instead?

**Comment by stwalkerster** [comment]
Re-opening as this clearly has not worked in prod. :(

**Comment by Dane2007** [comment]
This may be a naive question and I may be missing something but... Why are we using the password type for 2FA instead of a regular text box?  I don't know of many places that do that since it's just numbers that cycle every 30 seconds.

**Comment by stwalkerster** [comment]
I think I configured it that way because it was a credential which could
be shoulder-surfed.

TOTP credentials are almost always valid for more than 30 seconds - a
hard cut-off at the end of the window is almost certain to cause
occasional angst if they try submitting a token towards the end of the
window, even assuming the phone and server clocks are in sync.


One thing to be aware of though is that the YubiKey OTP auth has the
*potential* to be reusable in the scenario where the validation server
is self-hosted. In theory, this means that if you know of another server
which is not using the Yubico-hosted server, then capturing an OTP could
lead to it's successful reuse elsewhere. The YubiKey OTP process has a
sequence number which is stored as a last-seen value, and the validation
server will refuse to validate any OTP which has a lower sequence number
than it's last seen value - but different validation servers will have
different "last seen" values stored.

It's worth noting, however, that the opportunity for shoulder-surfing an
OTP value from TOTP is however long it takes for the user to look up the
code, and copy or type it in, and the window in which they have to do
that is 2m 30s (two window durations either side of the usual time). The
opportunity for shoulder-surfing a YubiKey OTP is the approximately 1
second it takes to autotype and submit the token, but the lifetime is
effectively indefinite, as it depends on the validation server seeing
this token before it's invalid.


tl;dr: I think it's safe to put it to type="text", though my last
attempt to avoid doing that but use browser hints worked locally, but
hasn't worked in prod for some reason. :(
